src/connectedk8s/azext_connectedk8s/_troubleshootutils.py

```diff
@@ -0,0 +1,802 @@
+        if storage_space_available:
+            with open(connected_cluster_resource_path, 'w+') as cc:
+                cc.write(str(connected_cluster))
+        return "Passed", storage_space_available
+
+    except OSError as e:
+        if "[Errno 28]" in str(e):
+            storage_space_available = False
+            telemetry.set_exception(exception=e, fault_type=consts.No_Storage_Space_Available_Fault_Type, summary="No space left on device")
+            shutil.rmtree(filepath_with_timestamp, ignore_errors=False, onerror=None)
+        else:
+            logger.warning("An expection has occured while trying to store the connected cluster resource logs from the cluster. Exception: {}".format(str(e)) + "\n")
+            telemetry.set_exception(exception=e, fault_type=consts.Connected_Cluster_Resource_Fault_Type, summary="Eror occure while storing the connected cluster resource logs")
+            cli_outputs.append("An expection has occured while trying to store the connected cluster resource logs from the cluster. Exception: {}".format(str(e)) + "\n")
+
+    except Exception as e:
+        logger.warning("An expection has occured while trying to store the connected cluster resource logs from the cluster. Exception: {}".format(str(e)) + "\n")
+        telemetry.set_exception(exception=e, fault_type=consts.Connected_Cluster_Resource_Fault_Type, summary="Eror occure while storing the connected cluster resource logs")
+        cli_outputs.append("An expection has occured while trying to store the connected cluster resource logs from the cluster. Exception: {}".format(str(e)) + "\n")
+
+    return "Failed", storage_space_available
+
+
+def arc_agents_logger(corev1_api_instance, filepath_with_timestamp, storage_space_available):
+
+    global cli_outputs
+
+    try:
+
+        if storage_space_available:
+            # To retrieve all of the arc agents pods that are presesnt in the Cluster
+            arc_agents_pod_list = corev1_api_instance.list_namespaced_pod(namespace="azure-arc")
+
+            # Traversing thorugh all agents
+            for each_agent_pod in arc_agents_pod_list.items:
+
+                # Fethcing the current Pod name and creating a folder with that name inside the timestamp folder
+                agent_name = each_agent_pod.metadata.name
+                arc_agent_logs_path = os.path.join(filepath_with_timestamp, "Arc_Agents_logs")
+                try:
+                    os.mkdir(arc_agent_logs_path)
+                except FileExistsError:
+                    pass
+
+                agent_name_logs_path = os.path.join(arc_agent_logs_path, agent_name)
+                try:
+                    os.mkdir(agent_name_logs_path)
+                except FileExistsError:
+                    pass
+
+                # If the agent is not in Running state we wont be able to get logs of the containers
+                if(each_agent_pod.status.phase != "Running"):
+                    continue
```

Comment:
If the pod is not in running state, we should display a warning to the user that one of their pods is not in running state and the name of the pod that's not in running state.